kansa/lib/purchase.js

```diff
@@ -257,7 +255,10 @@ class Purchase {
     }).then(_upgrades => {
       upgrades = _upgrades;
       const newMemberPaymentItems = newMembers.map(p => {
-        const mp = prices[p.data.membership] || 0
+        const mp = prices[p.data.membership]
+        if (typeof mp !== 'number' || mp < 0) {
+          throw new InputError(`Membership type not available for purchase: ${JSON.stringify(p.data.membership)}`)
+        }
```

Comment:
This method is getting really large. Would you consider breaking up into more functions?